### DISCUSSION TITLE
I do not understand how to connect my server to my gate server

### DISCUSSION THREAD

**Discussion by joehil** [e154/smart-home#266] [Q&A]
Hello Alexey,

I am sorry to have a question again: When my server starts, it will always connect to your sample gate server. I do not understand how I can end this behaviour.

How can I configure the server to connect to my gate server? 

Thank you very much!

With kindest regards
Joerg

**Comment by e154** [comment]
Update the Server Host field in the settings. To do this, start the server in super user mode (root_mode => true), or enter sudo -p [password] in the web console (root_secret from the configuration file)

Before starting the gate, edit the config.gate.json file. start the server with the gate flag.

**Reply by joehil** [reply]
Hello Alexey,

unfortunately I do not quite understand what you mean and setting the root_mode to true in config.json does not help. Do you have further hints for me?

With kindest regards
Joerg

**Reply by e154** [reply]
Hello Joerg,

Let's start in order, does the gate server work?
Does he have a white IP address?
Did you check the configuration file config.gate.json before launching?

**Reply by joehil** [reply]
Hello Alexey,

as I am not sure what you mean by "white IP address" I show you some data:

This is me config.gate.json:
{
  "api_http_port": 8080,
  "api_debug": true,
  "api_gzip": true,
  "pprof": false,
  "domain": "localhost",
  "https": false,
  "proxy_timeout": 5,
  "proxy_idle_timeout": 10,
  "proxy_secret_key": ""
}

This is the output from the gate server:
INFO    gate                    server/server.go:66 >   debug enabled
INFO    gate                    server/gate_server.go:93 >      Started ...
INFO    gate                    server/server.go:117 >  server started at :8080

The gate server runs on the same physical node as the server, and they are both in a local network (192.168.*). 

With kindest regards
Joerg

**Reply by e154** [reply]
Hello Joerg,

The following settings should be suitable for the above configuration, for example IP address 192.168.0.1.

<img width="721" alt="img" src="https://github.com/e154/smart-home/assets/11819743/7e0feaf1-fa3a-4efe-b6fe-d59bf396766f">

But in order for the gateway server to work and for you to be able to connect to it from the Internet, you need to be able to access your physical server from the Internet, have an external IP address issued by your ISP (white IP address), or install a gateway server in the cloud.

As a temporary solution, you can configure a connection to the current gate server. understand and figure out whether a “smart home” is right for you or not. Only those who know the “Client ID” will have access to your server.

**Reply by e154** [reply]
![example of network drawio](https://github.com/e154/smart-home/assets/11819743/fc3740ae-eef7-44e7-80fe-06a54a1a498c)

servers can be located on the same subnet, but if the gate server does not have an external address, then the meaning of the implementation is lost.

**Reply by joehil** [reply]
Hello Alexey,

I am sorry, but your software is probably not the right solution for me. I liked the idea to have a home automation solution written in golang (a language that I like a lot) and perhaps get the opportunity to contribute.

But it seems that I really do not understand the architecture of your software. I have been trying to make it work for several weeks now, but I do not understand how it works.

So thank you very much for your support and good luck for you and your product.

With kindest regards
Joerg

**Reply by e154** [reply]
Hello Joerg,

Thank you for trying out our Smart Home project. We understand that there might be challenges at the beginning, and we'd like to offer our assistance and additional resources to make the onboarding process smoother.

If you have specific questions or uncertainties, please feel free to reach out. We are ready to provide additional support and help you understand and configure the system. Perhaps together, we can overcome any difficulties you've encountered.

If you ever decide to revisit the idea of using Smart Home in the future, we'll be here, ready to assist.

Best regards,
Alex
